File: app/src/main/java/nz/co/codebros/quakesnz/ui/FeatureDetailActivityModule.java
package nz.co.codebros.quakesnz.ui;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import nz.co.codebros.quakesnz.detail.QuakeDetailFragment;
import nz.co.codebros.quakesnz.detail.QuakeDetailModule;
import nz.co.codebros.quakesnz.map.QuakeMapFragment;
import nz.co.codebros.quakesnz.module.FeatureModule;

/**
 * Created by leandro on 3/07/17.
 */
@Module
public abstract class FeatureDetailActivityModule {

    @ContributesAndroidInjector(modules = {
            QuakeDetailModule.class,
            FeatureModule.class
    })
    abstract QuakeDetailFragment quakeDetailFragment();

    @ContributesAndroidInjector(modules = FeatureModule.class)
    abstract QuakeMapFragment quakeMapFragment();
}

File: app/src/main/java/nz/co/codebros/quakesnz/ui/FeatureDetailActivityModule.kt
package nz.co.codebros.quakesnz.ui

import dagger.Module
import dagger.android.ContributesAndroidInjector
import nz.co.codebros.quakesnz.detail.QuakeDetailFragment
import nz.co.codebros.quakesnz.detail.QuakeDetailModule
import nz.co.codebros.quakesnz.map.QuakeMapFragment
import nz.co.codebros.quakesnz.module.FeatureModule

/**
 * Created by leandro on 3/07/17.
 */
@Module
abstract class FeatureDetailActivityModule {

    @ContributesAndroidInjector(modules = arrayOf(QuakeDetailModule::class, FeatureModule::class))
    internal abstract fun quakeDetailFragment(): QuakeDetailFragment

    @ContributesAndroidInjector(modules = arrayOf(
            QuakeMapFragment.Module::class,
            FeatureModule::class
    ))
    internal abstract fun quakeMapFragment(): QuakeMapFragment
}
